src/components/Album.js

```diff
@@ -30,82 +35,332 @@ import AddToWishlistButton from './layout/AddToWishlistButton/AddToWishlistButto
+  const [formData, setFormData] = useState({
+    title: '',
+    comment: '',
+    rating: '',
+  });
+  const [isEdit, setIsEdit] = useState('');
+  const [editObj, setEditObj] = useState('');
+  const [loginID, setLoginID] = useState('');
 
-  const handleOpenDialog = () => {
-    setOpenDialog(true);
+  const { status } = useAuth();
+  const isLoggedIn = status === AuthStatus.LoggedIn;
+
+  console.log('isLoggedIn=====', isLoggedIn)
+
+  const handleOpenDialog = async () => {
+    try {
+      setOpenDialog(true);
+      const response = await axiosInstance.get(`/reviews/album/${album._id}`);
+      setReviewList(response.data.allProductReviews);
+    } catch (error) {
+      console.error('Error fetching review:', error);
+    }
   };
 
   const handleCloseDialog = () => {
     setOpenDialog(false);
+    setIsEdit('');
+    setEditObj('');
+    setFormData({
+      title: '',
+      comment: '',
+      rating: '',
+    });
+  };
+
+  const show = () => {
+    const data = reviewList.filter((val) => val.user === loginID);
+    return data.length === 0 && isLoggedIn;
+  };
+
+  const handleAddReview = async () => {
+    try {
+      const data = {
+        ...formData,
+        user: loginID,
+        album: album._id,
+      };
+
+      const response = await axiosInstance.post(
+        `/reviews/album/${album._id}`,
+        data
+      );
+      setReviewList([...reviewList, response.data.review]);
+      setFormData({
+        title: '',
+        comment: '',
+        rating: '',
+      });
+    } catch (error) {
+      console.error('Error creating review:', error);
+    }
+  };
+
+  const handleDeleteReview = async (ID) => {
+    try {
+      await axiosInstance.delete(`/reviews/${ID}`);
+
+      const deleteData = reviewList.filter((val) => val._id !== ID);
+      setReviewList(deleteData);
+    } catch (error) {
+      console.error('Error delete review:', error);
+    }
+  };
+
+  const handleEditReview = async () => {
+    try {
+      const editData = {
+        title: editObj.title,
+        comment: editObj.comment,
+        rating: editObj.rating,
+      };
+
+      const editTest = {
+        title: editObj.title,
+        comment: editObj.comment,
+        rating: editObj.rating,
+        user: editObj.user,
+        album: editObj.album,
+      };
+
+      await axiosInstance.patch(`/reviews/${editObj._id}`, editData);
+
+      const data = reviewList.map((val) => {
+        if (val._id === editObj._id) {
+          val.title = editObj.title;
+          val.comment = editObj.comment;
+          val.rating = editObj.rating;
+        }
+        return val;
+      });
+
+      setReviewList(data);
+      setIsEdit('');
+    } catch (error) {
+      console.error('Error update review:', error);
+    }
   };
 
+  useEffect(() => {
+    if(isLoggedIn){
+      const ID = JSON.parse(localStorage.getItem('auth_user'));
+      ID.userId ? setLoginID(ID.userId) : setLoginID(ID.user.id);
+    }
+  }, []);
```

Comment:
Instead of this; we can also just use the `useAuth` hook to get the user as it will also store that data:

```suggestion
  const { user } = useAuth
  const loginID = user?.user?.id
```

In which case we don't need to have a `useState` variable for loginID since it can just be derived from the value of that hook so line 46 can be deleted